helm-ag.el

```diff
@@ -637,14 +637,16 @@ Default behaviour shows finish and result in mode-line."
         (if (not (file-writable-p file))
             (cl-incf read-only-files)
           (if lines-list
-              (push (list line body ovs) lines-list)
+              (progn
+                (push (list line body ovs) lines-list)
+                (puthash file lines-list files-to-lines))
```

Comment:
In original code, updated value is not stored.

Reply:
Ah lisp list...needed to update the map to point to the new head of the list.